tests/components/blebox/test_config_flow.py

```diff
@@ -0,0 +1,192 @@
+"""Test Home Assistant config flow for BleBox devices."""
+
+import blebox_uniapi
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.blebox import config_flow
+from homeassistant.setup import async_setup_component
+
+from .conftest import mock_config, mock_only_feature, setup_product_mock
+
+from tests.async_mock import DEFAULT, AsyncMock, PropertyMock, patch
+
+
+def create_valid_feature_mock(path="homeassistant.components.blebox.Products"):
+    """Return a valid, complete BleBox feature mock."""
+    feature = mock_only_feature(
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateBox-1afe34db9437-0.position",
+        full_name="gateBox-0.position",
+        device_class="gate",
+        state=0,
+        async_update=AsyncMock(),
+        current=None,
+    )
+
+    product = setup_product_mock("covers", [feature], path)
+
+    type(product).name = PropertyMock(return_value="My gate controller")
+    type(product).model = PropertyMock(return_value="gateController")
+    type(product).type = PropertyMock(return_value="gateBox")
+    type(product).brand = PropertyMock(return_value="BleBox")
+    type(product).firmware_version = PropertyMock(return_value="1.23")
+    type(product).unique_id = PropertyMock(return_value="abcd0123ef5678")
+
+    return feature
+
+
+@pytest.fixture
+def valid_feature_mock():
+    """Return a valid, complete BleBox feature mock."""
+    return create_valid_feature_mock()
+
+
+@pytest.fixture
+def flow_feature_mock():
+    """Return a mocked user flow feature."""
+    return create_valid_feature_mock(
+        "homeassistant.components.blebox.config_flow.Products"
+    )
+
+
+async def test_flow_works(hass, flow_feature_mock):
+    """Test that config flow works."""
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)